### ISSUE TITLE
Port CMake scripts to play nice on conda-forge

### ISSUE THREAD

**Issue by kordejong** [computationalgeography/lue#425]
Currently we need to hack things in to get the conda-forge build going. Try to upstream these things zo we don't need to do it ever again.

**Comment by kordejong** [comment]
@OliverSchmitz This is a ticket to dump all things related to the build of the conda package.

**Comment by OliverSchmitz** [comment]
```
# This must be set before project()
set(CMAKE_OSX_DEPLOYMENT_TARGET "10.15" CACHE STRING "Minimum OS X deployment version")
```
otherwise filesystem will fail to compile

**Comment by OliverSchmitz** [comment]
Perhaps don't do the CMAKE_OSX_DEPLOYMENT_TARGET yet

**Comment by OliverSchmitz** [comment]
Please avoid the need to set LD_PRELOAD to run the python unit tests

**Comment by kordejong** [comment]
Most things mentioned here are being fixed as part of #420
